src/lib/operators/union_positions.cpp

```diff
@@ -225,7 +225,7 @@ std::shared_ptr<const Table> UnionPositions::_prepare_operator() {
    */
   const auto add = [&](const auto& table) {
     auto current_pos_list = std::shared_ptr<const PosList>();
-    const auto first_chunk = table->get_chunk(ChunkID{0});
+    const auto& first_chunk = table->get_chunk(ChunkID{0});
```

Comment:
Why don't you check for nullptr in this file?

Reply:
Because of this Assert:
```cpp
/**
   * Both tables must contain only ReferenceSegments
   */
  Assert(input_table_left()->type() == TableType::References && input_table_right()->type() == TableType::References,
         "UnionPositions doesn't support non-reference tables yet");
```